src/System.Windows.Forms/src/System/Windows/Forms/TrackBar.TrackBarAccessibleObject.cs

```diff
@@ -0,0 +1,185 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Drawing;
+using static Interop;
+
+namespace System.Windows.Forms
+{
+    public partial class TrackBar
+    {
+        internal class TrackBarAccessibleObject : ControlAccessibleObject
+        {
+            private readonly TrackBar _owningTrackBar;
+            private TrackBarFirstButtonAccessibleObject? _firstButtonAccessibleObject;
+            private TrackBarLastButtonAccessibleObject? _lastButtonAccessibleObject;
+            private TrackBarThumbAccessibleObject? _thumbAccessibleObject;
+
+            public TrackBarAccessibleObject(TrackBar owningTrackBar) : base(owningTrackBar)
+            {
+                _owningTrackBar = owningTrackBar ?? throw new ArgumentNullException(nameof(owningTrackBar));
+            }
+
+            public override Rectangle Bounds
+            {
+                get
+                {
+                    if (!_owningTrackBar.IsHandleCreated)
+                    {
+                        return Rectangle.Empty;
+                    }
+
+                    int left = 0;
+                    int top = 0;
+                    int width = 0;
+                    int height = 0;
+                    GetSystemIAccessibleInternal()?.accLocation(out left, out top, out width, out height, GetChildId());
+
+                    return new Rectangle(left, top, width, height);
+                }
+            }
+
+            public override string DefaultAction => _owningTrackBar.AccessibleDefaultActionDescription;
+
+            public override AccessibleRole Role
+                => Owner.AccessibleRole != AccessibleRole.Default
+                    ? Owner.AccessibleRole
+                    : AccessibleRole.Slider;
+
+            /// <summary>
+            ///  Gets the accessible state.
+            /// </summary>
+            public override AccessibleStates State
+                => GetSystemIAccessibleInternal()?.get_accState(GetChildId()) is object accState
+                    ? (AccessibleStates)accState
```

Comment:
Would this work?
```suggestion
                => GetSystemIAccessibleInternal()?.get_accState(GetChildId()) is AccessibleStates accState
                    ? accState
```

If works, the same below

Reply:
It doesn't work because we are getting an `int`, not a `AccessibleStates`. We have to forcibly cast the int to the `AccessibleStates`.